calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp

```diff
@@ -1249,134 +1398,107 @@ std::pair<double, double> ExtrinsicReflectorBasedCalibrator::computeCalibrationE
+                            << initial_radar_to_lidar_eigen_.matrix());
+    RCLCPP_INFO_STREAM(
+      this->get_logger(), "Pure rotation calibration radar->lidar transform:\n"
+                            << calibrated_radar_to_lidar_transformation.matrix());
+  } else if (transformation_type_ == TransformationType::svd_2d) {
+    std::tie(lidar_points_ocs_, radar_points_rcs_) = getPointsSet();
+    estimator.setPoints(lidar_points_ocs_, radar_points_rcs_);
+    estimator.estimateSVDTransformation(transformation_type_);
+    calibrated_radar_to_lidar_transformation = estimator.getTransformation();
+    RCLCPP_INFO_STREAM(
+      this->get_logger(), "Initial radar->lidar transform:\n"
+                            << initial_radar_to_lidar_eigen_.matrix());
+    RCLCPP_INFO_STREAM(
+      this->get_logger(), "2D calibration radar->lidar transform:\n"
+                            << calibrated_radar_to_lidar_transformation.matrix());
+  } else if (
+    transformation_type_ == TransformationType::svd_3d ||
+    transformation_type_ == TransformationType::zero_roll_3d) {
+    std::tie(lidar_points_ocs_, radar_points_rcs_) = getPointsSet();
+    estimator.setPoints(lidar_points_ocs_, radar_points_rcs_);
 
-  calibrated_2d_radar_to_radar_parallel_transformation.translation().z() =
-    (initial_radar_to_lidar_eigen_ * radar_parallel_to_lidar_eigen_.inverse()).translation().z();
-  Eigen::Isometry3d calibrated_2d_radar_to_lidar_transformation =
-    calibrated_2d_radar_to_radar_parallel_transformation * radar_parallel_to_lidar_eigen_;
+    if (transformation_type_ == TransformationType::zero_roll_3d)
+      estimator.estimateZeroRollTransformation();
+    else if (transformation_type_ == TransformationType::svd_3d)
+      estimator.estimateSVDTransformation(transformation_type_);
 
-  // Estimate the 2D transformation estimating only yaw
-  double delta_cos = delta_cos_sum / converged_tracks_.size();
-  double delta_sin = -delta_sin_sum / converged_tracks_.size();
+    calibrated_radar_to_lidar_transformation = estimator.getTransformation();
+    RCLCPP_INFO_STREAM(
+      this->get_logger(), "Initial radar->lidar transform:\n"
+                            << initial_radar_to_lidar_eigen_.matrix());
+    RCLCPP_INFO_STREAM(
+      this->get_logger(), "3D calibration radar->lidar transform:\n"
+                            << calibrated_radar_to_lidar_transformation.matrix());
+  }
 
-  Eigen::Matrix3d delta_rotation;
-  delta_rotation << delta_cos, -delta_sin, 0.0, delta_sin, delta_cos, 0.0, 0.0, 0.0, 1.0;
-  Eigen::Isometry3d delta_transformation = Eigen::Isometry3d::Identity();
-  delta_transformation.linear() = delta_rotation;
-  Eigen::Isometry3d calibrated_rotation_radar_to_lidar_transformation =
-    delta_transformation * initial_radar_to_lidar_eigen_;
+  calculateCalibrationError(calibrated_radar_to_lidar_transformation);
+}
 
+void ExtrinsicReflectorBasedCalibrator::calculateCalibrationError(
```

Comment:
`calculateCalibrationError` and `computeCalibrationError` exist in this file with similar yet different purposes and signatures. This makes it very difficult to read. Choose one and overload, or use different more appropriate words

Reply:
The current `calculateCalibrationError` is unnecessarily deep in the call hierarchy.
In code metrics, the less deep the better.

Semantically speaking it should be between:
```
estimateTransformation();
<-- 
crossValEvaluation();
```

Reply:
Thanks! fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/c1a1ea3e4ea4d90e057ab97497431000437e3acb